tests/test_eepromfs.c

```diff
@@ -0,0 +1,83 @@
+    };
+
+    int result;
+
+    result = eepfs_init(eeprom_files, 2);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs init failed");
+    DEFER(eepfs_close());
+
+    if (!eepfs_verify_signature()) {
+        LOG("EEPROM signature is not valid;\nerasing all data!\n");
+        eepfs_wipe();
+    }
+
+    // Test reading zeroed-out files from wiped EEPROM
+    result = eepfs_read("file1", file1_dst);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs read failed");
+    result = memcmp(file1_src, file1_dst, sizeof(file1_src));
+    ASSERT_EQUAL_SIGNED(result, 0, "eepfs erase/read mismatch");
+    result = eepfs_read("file2", file2_dst);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs read failed");
+    result = memcmp(file2_src, file2_dst, sizeof(file2_src));
+    ASSERT_EQUAL_SIGNED(result, 0, "eepfs erase/read mismatch");
+
+    // Test writing and reading file1
+    for (int i = 0; i < sizeof(file1_src); i++) {
+        file1_src[i] = i;
+    }
+    result = eepfs_write("file1", file1_src);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs write failed");
+    result = eepfs_read("file1", file1_dst);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs read failed");
+    result = memcmp(file1_src, file1_dst, sizeof(file1_src));
+    ASSERT_EQUAL_SIGNED(result, 0, "eepfs write/read mismatch");
+
+    // Test erasing file1
+    result = eepfs_erase("file1");
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs erase failed");
+    memset(file1_src, 0, sizeof(file1_src));
+    result = eepfs_read("file1", file1_dst);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs read failed");
+    result = memcmp(file1_src, file1_dst, sizeof(file1_src));
+    ASSERT_EQUAL_SIGNED(result, 0, "eepfs erase/read mismatch");
+
+    // Test writing and reading file2
+    for (int i = 0; i < sizeof(file2_src); i++) {
+        file2_src[i] = i;
+    }
+    result = eepfs_write("file2", file2_src);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs write failed");
+    result = eepfs_read("file2", file2_dst);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs read failed");
+    result = memcmp(file2_src, file2_dst, sizeof(file2_src));
+    ASSERT_EQUAL_SIGNED(result, 0, "eepfs write/read mismatch");
+
+    // Ensure file1 was not modified
+    result = eepfs_read("file1", file1_dst);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs read failed");
+    result = memcmp(file1_src, file1_dst, sizeof(file1_src));
+    ASSERT_EQUAL_SIGNED(result, 0, "eepfs erase/read mismatch");
+
+    // Test erasing file2
+    result = eepfs_erase("file2");
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs erase failed");
+    memset(file2_src, 0, sizeof(file2_src));
+    result = eepfs_read("file2", file2_dst);
+    ASSERT_EQUAL_SIGNED(result, EEPFS_ESUCCESS, "eepfs read failed");
+    result = memcmp(file2_src, file2_dst, sizeof(file2_src));
+    ASSERT_EQUAL_SIGNED(result, 0, "eepfs erase/read mismatch");
```

Comment:
Maybe at the end you can initialize a different filesystem and check that the signature fails? So that we cover also that codepath.

Reply:
https://github.com/DragonMinded/libdragon/pull/125/commits/fd04a6359d93cd5f1d34906a97b287301ab69b55